commands/history/export.go

```diff
@@ -62,6 +63,25 @@ func runExport(ctx context.Context, dockerCli command.Cli, opts exportOptions) e
 			return errors.Errorf("no record found for ref %q", ref)
 		}
 
+		if opts.finalize {
+			var finalize bool
+			for _, rec := range recs {
+				if rec.Trace == nil {
+					finalize = true
+					break
+				}
+			}
+			if finalize {
+				recs, err = queryRecords(ctx, ref, nodes, &queryOptions{
```

Comment:
This would only work if ref is set. Might need to call finalize directly for all the refs in here where `Trace==nil` and then to full `queryRecords()` again.